src/qilib/data_set/data_set.py

```diff
@@ -116,7 +116,8 @@ def add_data(self, index_or_slice: Union[Tuple[int, ...], int], data: Dict[str,
         """
 
         for array_name, data_value in data.items():
-            self._data_arrays[array_name][index_or_slice] = data_value
+            getattr(self, array_name)[index_or_slice] = data_value
```

Comment:
This exposes more attributes then it should. Now this "works:" `data_set.add_data(0, {'storage': 22})` but breaks the data set as storage is now the int 22.